avocado/utils/network/interfaces.py

```diff
@@ -144,16 +144,37 @@ def bring_up(self):
         except Exception as ex:
             raise NWException("Failed to bring up: %s" % ex)
 
-    def is_link_up(self):
-        """Check if the interface is up or not.
+    def is_admin_link_up(self):
+        """Check the admin link state is up or not.
 
-        :return: True or False. True if the current state is UP, otherwise
-                 will return False.
+        :return: True or False, True if network interface state is 'up'
+                 or 'unknown' otherwise will return False.
         """
-        if self.get_link_state() == 'up':
+        if self.get_link_state() == 'up' or 'unknown':
```

Comment:
@smruti77 get_link_state() is actually checking for an operational state why you are calling in is_admin_link_up ?
also self._get_interface_details() is acutually checking the admin state than why you are calling in is_operational_link_up ?  is this a typo ?  it should be other way round

@beraldoleal we need to consider unknown state while checking veth interface due to this limitation from https://www.suse.com/support/kb/doc/?id=000019147

Reply:
@abdhaleegit I understand that we need to check for the unknown state, but as I mention before and you noticed too, we need to this in the proper place to avoid confusion.